scrapy/core/scheduler.py

```diff
@@ -256,6 +270,10 @@ def next_request(self) -> Optional[Request]:
         Increment the appropriate stats, such as: ``scheduler/dequeued``,
         ``scheduler/dequeued/disk``, ``scheduler/dequeued/memory``.
         """
+        delayed_request = self.dpqs.pop()
+        if delayed_request:
+            self.stats.inc_value('scheduler/dequeued/delayed/memory', spider=self.spider)
+            self._enqueue_request(delayed_request)
```

Comment:
I am thinking that it may be best to pop `request_delay` from requests before enqueuing them back, so that if they are re-scheduled for whatever reason the same delay is not applied again.